homeassistant/helpers/intent.py

```diff
@@ -243,6 +243,19 @@ class MatchTargetsConstraints:
     allow_duplicate_names: bool = False
     """True if entities with duplicate names are allowed in result."""
 
+    @property
+    def has_constraints(self) -> bool:
+        """Returns True if at least one constraint is set (ignores assistant)."""
+        return (
+            bool(self.name)
+            or bool(self.area_name)
+            or bool(self.floor_name)
+            or bool(self.domains)
+            or bool(self.device_classes)
+            or bool(self.features)
+            or bool(self.states)
+        )
```

Comment:
```suggestion
        return bool(
            self.name
            or self.area_name
            or self.floor_name
            or self.domains
            or self.device_classes
            or self.features
            or self.states
        )
```

Reply:
This is what I did and the type checker complained.